From 2c0413f849b9480e7fd2eb7f36d0fb13fca75a3d Mon Sep 17 00:00:00 2001
From: Arie Brown <abrown@habana.ai>
Date: Tue, 01 Feb 2022 12:08:36 +0200
Subject: [PATCH] [SW-72831] gaudi2: uart: print device id

similar to gaudi, for making deubgging easier - add the option to
retrieve device id on preboot uart.

Signed-off-by: Arie Brown <abrown@habana.ai>
Change-Id: I5c067d1f989ec6fa3d227b7ba4c937c5b5fd6131
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index aa1a49b..9f42a2c 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -369,3 +369,18 @@
 	regmap_writel(0, mmCPU_MSTR_IF_AXUSER_LB_LOCK);
 	regmap_writel(LOCK_BIT_OVRD_MASK, mmCPU_MSTR_IF_AXUSER_LB_OVRD);
 }
+
+#define HL_GAUDI2_MEZZ_ID_SHIFT	2
+#define HL_GAUDI2_MEZZ_ID_MASK	0x1C /* GENMASK(4, 2), use 3 out of 5 */
+
+void hl_esc_key_handle(void)
+{
+	u8 mezz_id;
+	u32 strap;
+
+	strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H);
+	mezz_id = (strap >> HL_GAUDI2_MEZZ_ID_SHIFT) & HL_GAUDI2_MEZZ_ID_MASK;
+	preboot_puts_raw("OAM[");
+	preboot_puth_raw(mezz_id, false);
+	preboot_puts("] Gaudi2!\n");
+}
